rp2040-hal/src/pwm.rs

```diff
@@ -88,7 +87,7 @@ impl $PXi {
     // will instead receive a single gpio pin.
     fn init_io(&self, pwm: &mut pac::$PWMX, pad : &mut pac::PADS_BANK0, io : &mut pac::IO_BANK0, resets: &mut pac::RESETS) -> () {
         //TODO: Merge these into gpio.rs split function after GPIO refactor. At the moment, this is here because these need to be reset for
-        // the PWM to work. However, because they're here, they'll be reset every time a new PWM pin is created (which is bad). Definitely needs to change once GPIO is redone.
+        // the PWM to work. However, because they're here, they'll be reset every time a new PWM pin is created (BAD).
         pwm.reset_bring_up(resets);
```

Comment:
This actually shouldn't reset them a second time. `reset_bring_up` only brings the peripheral out of reset. Should be a no-op (and a waste of a few cycles) if the peripheral has already been brought up.